lib/rex/ui/text/table.rb

```diff
@@ -330,13 +382,17 @@ def row_to_s(row) # :nodoc:
   # some text and a column index.
   #
   def pad(chr, buf, colidx, use_cell_pad = true) # :nodoc:
-    remainder = colprops[colidx]['MaxWidth'] - buf.length
-    val       = chr * remainder;
+    # Ensure we pad the minimum required amount
+    max = colprops[colidx]['MaxChar'] || colprops[colidx]['MaxWidth']
+    max = colprops[colidx]['MaxWidth'] if max.to_i > colprops[colidx]['MaxWidth'].to_i
+    remainder = max - buf.length
+    remainder = 0 if remainder < 0
```

Comment:
Agree, but this reads cleaner IMO - Ruby newcomers may have some difficulty with ternary operators (keep seeing it confuse new devs like asking them to find keys to the Humvee). Can change it if there's common desire of course.